src/Toml/Parser/Value.hs

```diff
@@ -170,10 +171,23 @@ doubleP :: Parser Double
 doubleP = lexeme (signed sc (num <|> inf <|> nan)) <?> "double"
   where
     num, inf, nan :: Parser Double
-    num = float
+    num = floatP
     inf = 1 / 0 <$ string "inf"
     nan = 0 / 0 <$ string "nan"
 
+floatP :: Parser Double
+floatP = check =<< readMaybe . concat <$> sequence [ digits, expo <|> dot ]
+  where
+    check = maybe (fail "Not a float") return
```

Comment:
Am I understand this correctly that this `fail` will be handled by `<|>`? I just don't know how it's implemented in `megaparsec`, does it use `fail` as well?

Reply:
I'm not too sure, because that situation should never happen. It's not a `megaparsec` function.

Reply:
Yes, I mean `fail` is somehow implemented for `Parser` from `megaparsec`. And we are using this `floatP` with `<|>` operator, so, that's why I'm asking, how it's going to be handled

Reply:
I just tried this:
```
ghci> Toml.Parser.Core.parse (fail "whoops" <|> boolP) "" "true"
Right True
```
so it's all good :)

Reply:
Yeah, let's not do `error`, `fail` is ok for now 👍

Reply:
I need to point out that `dateTimeP` has 2 `error` in it. Also `decimalP` has one, but I'm getting rid of it in #128